### PR TITLE
added netrc support

### PR DESCRIPTION
**Pull request by cansarigol**

### CODE REVIEW

File: httpx/config.py [hunk 1]
```diff
@@ -39,9 +39,16 @@ class SSLConfig:
     SSL Configuration.
     """
 
-    def __init__(self, *, cert: CertTypes = None, verify: VerifyTypes = True):
+    def __init__(
```
Comment (sethmlarson) [inline httpx/config.py]: Let's revert usages of trust_env for SSLConfig? The default behavior in the future should be use system trust stores unless we're specifically told which cert bundle to use.

Comment (cansarigol) [reply httpx/config.py]: You're right new approach for environment variables would be nice.
I reverted the last commit.

File: httpx/utils.py [hunk 2]
```diff
@@ -64,3 +66,12 @@ def guess_json_utf(data: bytes) -> typing.Optional[str]:
             return "utf-32-le"
         # Did not detect a valid UTF-32 ascii-range character
     return None
+
+
+def get_netrc_login(url: str) -> typing.Union[typing.Tuple[str, str, str], None]:
```
Comment (tomchristie) [inline httpx/utils.py]: Signature should either be `host` or `authority` here, right?
It's not clear to me from a glance at the netrc docs, which of those two is actually the correct thing to use?
Also we should prefer `typing.Optional[typing.Tuple[str, str, str]]` as the return annotation.

File: httpx/utils.py [hunk 3]
```diff
@@ -64,3 +66,12 @@ def guess_json_utf(data: bytes) -> typing.Optional[str]:
...
+    try:
+        _netrc = netrc.netrc(os.environ.get("DEFAULT_NETRC_PATH"))  # type: ignore
```
Comment (tomchristie) [inline httpx/utils.py]: I'd probably prefer a variable name that's not underscore prefixed here. `netrc_info` maybe?

File: tests/.netrc [hunk 4]
```diff
@@ -0,0 +1,3 @@
+machine test_entrc.httpx.com
```
Comment (tomchristie) [inline tests/.netrc]: Let's use the `example.org` domain, since it's there for this sort of thing.

File: tests/.netrc [hunk 5]
```diff
@@ -0,0 +1,3 @@
...
+login tomchristie
+password password123
```
Comment (tomchristie) [inline tests/.netrc]: I'd probably marginally prefer that we use a fake style username here `example-username` and `example-password` or something. Just feels marginally cleaner.

File: httpx/utils.py (hunk 3 above)
Comment (tomchristie) [reply httpx/utils.py]: Is `DEFAULT_NETRC_PATH` a standardized thing? (Or used by requests or something?)

File: tests/.netrc (hunk 4 above)
Comment (cansarigol) [reply tests/.netrc]: If appropriate, I used netrcexample.org. Because some tests are affected by example.org.

File: httpx/utils.py (hunk 3 above)
Comment (cansarigol) [reply httpx/utils.py]: I made it up to explain the initial path.

File: httpx/utils.py (hunk 2 above)
Comment (cansarigol) [reply httpx/utils.py]: `machine` in the config. I changed it as host I can change again if necessary

**Comment by cansarigol** [comment]
Thanks, I applied your comments.

File: httpx/utils.py (hunk 3 above)
Comment (sethmlarson) [reply httpx/utils.py]: I'm pretty sure `NETRC` is the env variable to specify the path

File: tests/client/test_auth.py [hunk 6]
```diff
@@ -67,3 +68,27 @@ def auth(request):
 
     assert response.status_code == 200
     assert response.json() == {"auth": "Token 123"}
+
+
+def test_entrc_auth():
```
Comment (StephenBrown2) [inline tests/client/test_auth.py]: Typo: `entrc` -> `netrc`

Comment (cansarigol) [reply tests/client/test_auth.py]: thanks, fixed

**Comment by tomchristie** [comment]
Great stuff, thanks!
Opening up two other issues that would act as further enhancements to this.

**Pull request merged**